spec/queries/matched_for_notification_spec.rb

```diff
@@ -59,17 +59,33 @@ def execute_query(field:, query_hash:)
       expect(lists).to include(@list5)
     end
 
+    it 'does not find subscriber list for all topics when not all topics present' do
+      lists = execute_query(field: :links, query_hash: { topics: ["uuid-234", "other-2"] })
+      expect(lists).not_to include(@list5)
+    end
+
     it 'finds subscriber lists matching any and all topics' do
       lists = execute_query(field: :links, query_hash: { topics: ["uuid-345", "uuid-678", "uuid-456"] })
       expect(lists).to include(@list6)
     end
 
+    it 'does not find subscriber list for mixture of all and any topics when not all topics present' do
+      lists = execute_query(field: :links, query_hash: { topics: ["uuid-345", "uuid-678"] })
+      expect(lists).not_to include(@list6)
+    end
+
     it 'finds subscriber lists matching a mix of all topics and policies' do
       lists = execute_query(field: :links, query_hash: { topics: ["uuid-345", "uuid-456", "other1"],
                                                          policies: ["uuid-567", "uuid-678", "other2"] })
       expect(lists).to include(@list7)
     end
 
+    it 'does not find subscriber list for mix of all topics and policies when not all policies present' do
+      lists = execute_query(field: :links, query_hash: { topics: ["uuid-345", "uuid-456", "other1"],
+                                                         policies: ["uuid-678", "other2"] })
+      expect(lists).not_to include(@list7)
+    end
```

Comment:
Good to have these tests 👍 but I'll make the same suggestion as I did on Oscar & Aga's PR: https://github.com/alphagov/email-alert-api/pull/867#discussion_r287263976

> Think these tests would be easier to read if they followed the Arrange, Act, Assert model. Currently the 'Arrange' all happens at the top and you have to scroll up and down to figure out what's going on.

Reply:
Maybe this could come with a later refactor?  Some of the lists are used in multiple tests, so we'd have to rethink the way the tests are broken down into smaller units.